### ISSUE TITLE
Add VILOCA results to dashboard

### ISSUE THREAD

**Issue by mcarrara-bioinfo** [GenSpectrum/cov-spectrum-website#813]
As initially discussed through other channels, CBG developed VILOCA to detect potential drug-resistance mutations in wastewater data and they wish is to add the results of this analysis to the cov-spectrum dashboard.

We have compiled [a document](https://docs.google.com/document/d/1ARxFO8A9MxNV3cvTnlMPplYAgVowY3eKTAnv4ZajI40/edit) with comments, the text that we wish to add and the figure to include. I also attached to this issue an example of the actual file to be shared and embedded.
The output of the VILOCA run will be directly the figure to embed in the page. 

The image is to be updated every time a VILOCA run is complete, which is likely to be with a slower frequency and at different time frames than the updates to the Swiss curves.

A discussion is necessary for the preferred method to exchange the figure.

[mutation_calls_on_potential_drug_resistance_sites.pdf](https://github.com/GenSpectrum/cov-spectrum-website/files/11676933/mutation_calls_on_potential_drug_resistance_sites.pdf)

**Comment by corneliusroemer** [comment]
The Google doc seems to not allow link-only viewing, is that on purpose?

**Comment by chaoran-chen** [comment]
Hi @mcarrara-bioinfo, I assume that it wouldn't be a problem to provide the plot as an SVG? I think that an SVG will be easier to embed on a website.

Should we maybe just create a new database table where you can store the content of the SVG? This should be the easiest because it is an already established method to exchange data.

**Comment by mcarrara-bioinfo** [comment]
> The Google doc seems to not allow link-only viewing, is that on purpose?

I have changed the access permissions, you should be able to access now.

> Hi @mcarrara-bioinfo, I assume that it wouldn't be a problem to provide the plot as an SVG? I think that an SVG will be easier to embed on a website.

I am confirming this with the developer, but I don't think that would be a problem.

> Should we maybe just create a new database table where you can store the content of the SVG? This should be the easiest because it is an already established method to exchange data.

On our side there is no preference on the exchange method as long as it's feasible from machines within the BSSE network and from Euler.

**Comment by chaoran-chen** [comment]
I just created the table `wastewater_viloca` which has a single column `plot_svg` (type: TEXT). Would that work? You should already have access to it.

**Comment by chaoran-chen** [comment]
The table has exactly one row (which currently contains "nothing here yet, please update!") and you can update it.

**Comment by mcarrara-bioinfo** [comment]
@chaoran-chen I have confirmed that the tool can be adapted to generate SVG files. I see the table and we can include code to push the SVG whenever new results are available.

I am currently working on the final tests, then it's a matter of waiting for a PR review on the CBG automation repository.

**Comment by mcarrara-bioinfo** [comment]
The tool has been adapted to generate SVG files. I attached an example.
We will start filling and updating the new table as soon as the automation that is supposed to run VILOCA is updated (code currently under review). At that point the heatmap can be published.

Question: in the current situation, will the heatmap be published automatically as soon as I fill the table or do you need to finalise something in the frontend?

![mutation_calls_on_potential_drug_resistance_sites](https://github.com/GenSpectrum/cov-spectrum-website/assets/19552826/045b1da5-4d95-4c7f-8594-c6be6089ae34)

**Comment by chaoran-chen** [comment]
No, we need to implement the integration into the frontend first. Could you please deposit a version of the chart so that we have something to work with?

**Comment by mcarrara-bioinfo** [comment]
I uploaded to the database a test heatmap you can use as reference

**Comment by mcarrara-bioinfo** [comment]
Delayed because of a much-needed refactoring of the automation that is supposed to generate the data and because of my holidays. I will resume working on it next week. I estimate everything to be ready to deploy by the beginning of October, considering testing and ironing out details after deploying the changes on our side.
I will check if I can start a non-automated run to build the most recent results, so that you can merge and close the issue earlier by publishing it while I finish with the automation. I will update you on this last point next week

**Comment by gordonkoehn** [comment]
Is this issue still ongoing, or has it been discarded? 

Resistance calls with VILOCA were never integrated into a web display, to my understanding, but are currently integrated into the regular email. 

Shall we close this? @DrYak?
